### PR TITLE
raise an exception when auxiliary failed at instance creation

### PR DESCRIPTION
**Pull request by BKaDamien**
- raise the brand new AuxiliaryCreationError exception when the aux instance creation failed
- remove unused timeout decorator
- add --fail-fast flag to cli

### CODE REVIEW

File: src/pykiso/cli.py [hunk 1]
```diff
@@ -173,6 +173,11 @@ def get_logging_options() -> LogOptions:
     default=None,
     help="allow the user to execute a subset of tests based on branch levels",
 )
+@click.option(
+    "--fail-fast",
```
Comment (sebclrsn) [inline src/pykiso/cli.py:177]: Maybe add the option flag in the same format as unittest (-f, --failfast)

File: src/pykiso/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+:module: exceptions
+
+:synopsis: Define all custom exceptions raised by pykiso framework
+
+.. currentmodule:: exceptions
+"""
+
+
+class PykisoError(Exception):
+    """General XCP specific exception used as basis for all others."""
```
Comment (sebclrsn) [inline src/pykiso/exceptions.py:20]: Docstring should be adapted

File: src/pykiso/exceptions.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    def __str__(self):
+        return self.message
+
+class AuxiliaryCreationError(PykisoError):
+    """Raised when an auxiliary instance creation failed,"""
```
Comment (sebclrsn) [inline src/pykiso/exceptions.py:26]:
```suggestion
    """Raised when an auxiliary instance creation failed."""
```

File: src/pykiso/test_coordinator/test_execution.py [hunk 4]
```diff
@@ -48,6 +49,7 @@ class ExitCode(enum.IntEnum):
     ONE_OR_MORE_TESTS_FAILED = 1
     ONE_OR_MORE_TESTS_RAISED_UNEXPECTED_EXCEPTION = 2
     ONE_OR_MORE_TESTS_FAILED_AND_RAISED_UNEXPECTED_EXCEPTION = 3
+    FAILED_AT_AUXILIARY_CREATION = 4
```
Comment (sebclrsn) [inline src/pykiso/test_coordinator/test_execution.py:52]:
```suggestion
    AUXILIARY_CREATION_FAILED = 4
```
Just to keep a similar format than the other members

**Review by sebclrsn — CHANGES_REQUESTED** [review]

File: src/pykiso/cli.py [hunk 5]
```diff
@@ -173,6 +173,11 @@ def get_logging_options() -> LogOptions:
     default=None,
     help="allow the user to execute a subset of tests based on branch levels",
 )
+@click.option(
```
Comment (sebclrsn) [inline src/pykiso/cli.py:176]: The CLI help in the readme and getting started should be updated with the new flag

File: src/pykiso/test_coordinator/test_execution.py (hunk 4 above)
Comment (BKaDamien) [reply src/pykiso/test_coordinator/test_execution.py:52]: ohh there is a format ???

File: src/pykiso/cli.py (hunk 1 above)
Comment (BKaDamien) [reply src/pykiso/cli.py:177]: as you want

File: src/pykiso/interfaces/thread_auxiliary.py [hunk 6]
```diff
@@ -205,9 +207,6 @@ def run(self) -> None:
             if not self.is_instance:
                 time.sleep(0.050)
 
-            # If auxiliary instance is created and is pausable
```
Comment (sebclrsn) [inline src/pykiso/interfaces/thread_auxiliary.py:208]: Not related to this, but just a few more findings regarding the cleanup (sorry I forgot to mention these last time):
- in auxiliary.py: docstrings of abcs create_instance and delete_instance are wrong, they should be something like `"""Handle auxiliary creation/deletion."""`
- in auxiliary.py: error log of suspend/resume can now be more specific, like `log.error(f"Auxiliary '{self}' is already running/stopped")`
- in thread_auxiliary and mp_auxiliary: attribute `self.wait_event` can be removed

**Review by sebastianpfischer — APPROVED** [review]

**Review by sebclrsn — APPROVED** [review]

**Comment by BKaDamien** [comment]
is it normal that since this morning this PR is block at some checks?

File: src/pykiso/cli.py [hunk 7]
```diff
@@ -173,6 +173,11 @@ def get_logging_options() -> LogOptions:
...
+    "--failfast",
+    is_flag=True,
+    help="stop the test run on the first error or failure",
```
Comment (Pog3k) [inline src/pykiso/cli.py:179]: maybe add an hint that default is True

**Review by Pog3k — COMMENTED** [review]
ok

**Pull request merged**